mne/encoding/model.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+from sklearn.base import BaseEstimator
+from scipy import linalg
+from ..externals.six import string_types
+
+
+class EncodingModel(object):
+    def __init__(self, est=None, delays=[0.], preproc_x=None, preproc_y=None):
+        """Fit a STRF model.
+
+        Fit an encoding model using time lags and a custom estimator.
+        Compatible either with continuous or events-based input
+        stimuli. It creates time lags for the input matrix, performs
+        preprocessing according to any transformers in the pipeline in `est`,
+        and then fits a model with the final estimator in `est`.
+
+        Parameters
+        ----------
+        est : instance of sklearn-style estimator
+            The estimator to use for fitting. This is any object that contains
+            a `fit` and `predict` method, which takes inputs of the form
+            (X, y), and which creates a `.coef_` attribute upon fitting.
+        delays : array, shape (n_delays,)
```

Comment:
But doesn't `decim` actually decimate the data? I'd prefer not to do that (aka, make the user decimate their data before the call to the modeling code). I think that if we allow for a tmin/tmax parameter here, then the same functionality could be accomplished, no? Or if we had a single parameter called `times` or `delays`, along with a flag `kind='window'` vs. `kind='custom'` or something? Then it could handle either a length 2 input for a window, or an arbitrary length input for custom delays.